lib/validation/validator.rb

```diff
@@ -1,132 +1,172 @@
 module Validation
-  module Rules
-    # A hash of rules for this object
-    def rules
-      @rules ||= {}
-    end
+  module Validator
+    attr_reader :object, :results
 
-    # A hash of errors for this object
-    def errors
-      @errors ||= {}
+    def initialize(object)
+      raise ArgumentError, "cannot validate nil" if object.nil?
```

Comment:
Validation rules are bound to specific fields of the object, not the object itself, so if a `nil` is given here, every rule will fail with `InvalidKey`. That's why I added this preemptive measure originally.

What would probably be better though is if we allowed `nil` here, and had a way to define rules that validate the object itself, and not one of its fields. So for now I've removed this exception to be able to add such a feature later without breaking backwards compatibility.